packages/web/app/components/PatientPrinting/modals/PatientPortalRegistrationModal.jsx

```diff
@@ -0,0 +1,155 @@
+import { TranslatedText } from '../../Translation/TranslatedText';
+import { BodyText } from '../../Typography';
+import { Button, OutlinedButton } from '../..';
+import { Colors } from '../../../constants';
+import { SendIcon } from '../../Icons/SendIcon';
+import { useRegisterPatientPortal } from '../../../api/mutations/useRegisterPatientPortalMutation';
+import { SendToPatientModal } from './SendToPatientModal';
+import { Box, Link } from '@mui/material';
+
+const StyledButtonRow = styled('div')`
+  display: flex;
+  align-items: center;
+  padding-inline: 1.75rem;
+  padding-block: 1.25rem;
+  gap: 1rem;
+  background-color: ${Colors.background};
+  border-top: 1px solid ${Colors.outline};
+`;
+
+const StyledQrCodeContainer = styled(Box)`
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  justify-content: center;
+  gap: 1.25rem;
+`;
+
+const generateQrCode = async registrationLink => {
+  return await QRCode.toDataURL(registrationLink);
+};
+
+const BottomRow = ({ onPrint, onSendToPatient, onClose }) => (
+  <StyledButtonRow>
+    <OutlinedButton onClick={onPrint}>
+      <TranslatedText
+        stringId="general.action.print"
+        fallback="Print"
+        data-testid="translatedtext-patient-portal-print"
+      />
+    </OutlinedButton>
+    <OutlinedButton
+      startIcon={<SendIcon htmlColor={Colors.primary} width={16} height={16} />}
+      style={{ marginLeft: 'auto' }}
+      onClick={onSendToPatient}
+    >
+      <TranslatedText
+        stringId="patientDetails.resources.patientPortalRegistration.modal.action.sendToPatient"
+        fallback="Send to patient"
+        data-testid="translatedtext-patient-portal-send-to-patient"
+      />
+    </OutlinedButton>
+    <Button onClick={onClose}>
+      <TranslatedText
+        stringId="general.action.close"
+        fallback="Close"
+        data-testid="translatedtext-patient-portal-close"
+      />
+    </Button>
+  </StyledButtonRow>
+);
+
+const QrCodeDisplay = ({ qrCode, registrationLink }) => {
+  if (!qrCode || !registrationLink) {
+    return null;
+  }
+
+  return (
+    <StyledQrCodeContainer>
+      <img src={qrCode} alt="Patient portal registration QR code" style={{ width: '11.5rem' }} />
+      <Link href={registrationLink} target="_blank">
+        {registrationLink}
+      </Link>
+    </StyledQrCodeContainer>
+  );
+};
+
+export const PatientPortalRegistrationModal = React.memo(() => {
+  const patient = useSelector(state => state.patient);
+
+  const [open, setOpen] = useState(true);
+  const [openSendToPatientModal, setOpenSendToPatientModal] = useState(false);
+  const [registrationLink, setRegistrationLink] = useState(null);
+  const [qrCode, setQrCode] = useState(null);
+
+  const { mutate: registerPatientPortal } = useRegisterPatientPortal({
+    onSuccess: data => {
+      setRegistrationLink(data?.registrationLink);
+      generateQrCode(data?.registrationLink).then(setQrCode);
+    },
+  });
+
+  useEffect(() => {
```

Comment:
Use effects can be confusing and I usually try to avoid them wherever possible. I think this one makes sense (it's a bit of a funny design) but it would be good to put a quick comment such as // Automatically register the patient for the portal when the modal opens